app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -160,7 +160,9 @@ public String getInitialUrl() {
     }
 
     private void updateURL(final String url) {
-        urlView.setText(UrlUtils.stripUserInfo(url));
+        if (!urlView.getText().toString().equals(url)){
```

Comment:
I guess it is possible for urlView.getText() to return null so !url.toString().equals(urlView.getText()) should be better, or a null check. I'm actually thinking of not changing this since branching may not be faster.